python/reg_interface/scripts/sca.py

```diff
@@ -8,37 +8,105 @@
 from reg_utils.reg_interface.common.sca_utils import *
 from reg_utils.reg_interface.common.sca_common_utils import *
 from time import *
-import array
-import struct
 import socket
 
-def main():
-
-    instructions = ""
-    ohMask = 0
-    ohList = []
-
-    if len(sys.argv) < 4:
-        print('Usage: sca.py <card_name> <oh_mask> <instructions>. If running on the card, put `local` instead of hostname')
-        print('instructions:')
-        print('  r:        SCA reset will be done')
-        print('  h:        FPGA hard reset will be done')
-        print('  hh:       FPGA hard reset will be asserted and held')
-        print('  fpga-id:  FPGA ID will be read through JTAG')
-        print('  sysmon:   Read FPGA sysmon data repeatedly')
-        print('  program-fpga:   Program OH FPGA with a bitfile or an MCS file. Requires a parameter "bit" or "mcs" and a filename')
+def compareFwFiles(args):
+    if ((args.fwFileBit is None) or (args.fwFileMCS is None)):
+        print("Usage: sca.py <cardName> <ohMask> compare-mcs-bit --fwFileMCS=<mcs_filename> --fwFileBit=<bit_filename>")
         return
+
+    import os
+    if "bit" not in args.fwFileBit:
+        print("you must supply a *.bit file to '--fwFileBit'")
+        exit(os.EX_USAGE)
+    if "mcs" not in args.fwFileMCS:
+        print("you must supply a *.mcs file to '--fwFileMCS'")
+        exit(os.EX_USAGE)
+
+    from reg_utils.reg_interface.arm.program_fpga import compare_mcs_bit
+    compare_mcs_bit(args.fwFileMCS, args.fwFileBit)
+    return
+
+def fpgaHardReset(args):
+    scaInit(args)
+    fpga_single_hard_reset()
+    return
+
+def fpgaHeldInHardReset(args):
+    ohList = scaInit(args)
+    fpga_keep_hard_reset(ohList)
+    return
+
+def fpgaId(args):
+    scaInit(args)
+    read_fpga_id(args.ohMask)
+    return
+
+def fpgaProgram(args):
```

Comment:
General question about this function, is there anything specific to the `bit`/`mcs` file treatment that couldn't be handled  by lines 61-66?
Reasoning: removing the option of `--fwFileBit` and `--fwFileMCS` and simply replacing it with a required argument `fwFile` would simplify the code

Reply:
Yes I agree.  I've implemented this change.